src/layouts/sidebar-layout/index.tsx

```diff
@@ -0,0 +1,44 @@
+import { PropsWithChildren, ReactNode } from 'react'
+// Layout
+import BaseLayout from 'layouts/base-layout'
+// Styles
+import s from './sidebar-layout.module.css'
+
+/**
+ * Renders a sidebar area alongside a main content area.
+ *
+ * The sidebar area is layed out as a sticky column on large viewports.
+ * When very tall content is provided, the sidebar area will scroll vertically.
```

Comment:
> * When very tall content is provided, the sidebar area will scroll vertically.

Copy-pasted the placeholder elements and scroll behaves nicely

<img width="1320" alt="CleanShot 2023-07-19 at 11 20 47@2x" src="https://github.com/hashicorp/dev-portal/assets/26389321/fab92e61-a28d-49f9-9db8-e3f1c9a88c56">